packages/tx/src/types.ts

```diff
@@ -93,20 +95,91 @@ export function isAccessList(input: AccessListBytes | AccessList): input is Acce
 
 /**
  * Encompassing type for all transaction types.
- *
- * Note that this also includes legacy txs which are
- * referenced as {@link Transaction} for compatibility reasons.
  */
-export type TypedTransaction =
-  | Transaction
-  | AccessListEIP2930Transaction
-  | FeeMarketEIP1559Transaction
-  | BlobEIP4844Transaction
+export enum TransactionType {
+  Legacy,
+  AccessListEIP2930,
+  FeeMarketEIP1559,
+  BlobEIP4844,
+}
+
+export interface Transaction {
+  [TransactionType.Legacy]: LegacyTransaction
+  [TransactionType.FeeMarketEIP1559]: FeeMarketEIP1559Transaction
+  [TransactionType.AccessListEIP2930]: AccessListEIP2930Transaction
+  [TransactionType.BlobEIP4844]: BlobEIP4844Transaction
+}
+
+export type UnknownTransaction =
+  | Transaction[TransactionType.Legacy]
+  | Transaction[TransactionType.AccessListEIP2930]
+  | Transaction[TransactionType.FeeMarketEIP1559]
+  | Transaction[TransactionType.BlobEIP4844]
+
+export type TransactionsArray = Array<UnknownTransaction>
```

Comment:
i think the `TypedTransaction[]` is more meaningful and code intuitive than `TransactionsArray` and would like to retain its usage even in the new refactored form

Reply:
Yes, would agree here.

Reply:
Renamed to TypedTransaction & TypedTransaction[]